src/main/java/org/opensearch/neuralsearch/search/query/HybridCollectorManager.java

```diff
@@ -149,9 +210,71 @@ public ReduceableSearchResult reduce(Collection<Collector> collectors) {
             TopDocsAndMaxScore topDocsAndMaxScore = new TopDocsAndMaxScore(newTopDocs, hybridTopScoreDocCollector.getMaxScore());
             return (QuerySearchResult result) -> { result.topDocs(topDocsAndMaxScore, getSortValueFormats(sortAndFormats)); };
         }
+
+        if (!hybridSortedTopDocCollectors.isEmpty()) {
+            HybridTopFieldDocSortCollector hybridSortedTopScoreDocCollector = hybridSortedTopDocCollectors.stream()
+                .findFirst()
+                .orElseThrow(() -> new IllegalStateException("cannot collect results of hybrid search query"));
+
+            HybridTopFieldDocSortCollector.SimpleFieldCollector simpleFieldCollector;
+            HybridTopFieldDocSortCollector.PagingFieldCollector pagingFieldCollector;
+            List<TopFieldDocs> topFieldDocs;
+            long maxTotalHits;
+            float maxScore;
+            if (hybridSortedTopScoreDocCollector instanceof HybridTopFieldDocSortCollector.SimpleFieldCollector) {
```

Comment:
I'm not sure you need whole if/else construct, seems that all three fields you're trying to get are defined at the abstract class level, so you don'r need to cast to the subclass